src/main/kotlin/com/infomaniak/core/utils/EmailUtils.kt

```diff
@@ -20,3 +20,20 @@ package com.infomaniak.core.utils
 import android.util.Patterns
 
 fun String.isValidEmail(): Boolean = Patterns.EMAIL_ADDRESS.matcher(trim()).matches()
+
+/**
+ * This regex should be closer to RFC5321 compared to Patterns.EMAIL_ADDRESS which is missing, amongst many things, the
+ * characters #&~!^`{}/=$*?|
+ *
+ * This regex comes from the iOS mail app that supports more email addresses than Android did with Patterns.EMAIL_ADDRESS
+ */
+private val rfc5321ValidEmailPattern =
+    "(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])".toRegex()
```

Comment:
Instead of using `\\` everywhere, can't we put the whole String between `"""` ?

Reply:
I thought about it too but I figured we might want to stick to the exact way iOS has written it instead to be able to more easily compare the two regex in case we ever need to change something in it. As it is, it's already enough of a monster so anything that would help maintain this code is worthwhile imo